--- c:\Users\trunet\Downloads\SoftwareSerial\icrmacros.h	2011-02-27 00:21:43.013000000 -0300
+++ c:\Users\trunet\Downloads\arduino-0022\libraries\SoftwareSerial\icrmacros.h	2011-02-27 00:19:44.322000000 -0300
@@ -35,6 +35,24 @@
 #define digitalPinToPCMSK(p)    (((p) <= 7) ? (&PCMSK2) : (((p) <= 13) ? (&PCMSK0) : (((p) <= 21) ? (&PCMSK1) : ((uint8_t *)0))))
 #define digitalPinToPCMSKbit(p) (((p) <= 7) ? (p) : (((p) <= 13) ? ((p) - 8) : ((p) - 14)))
 
+#elif defined(__AVR_ATmega644__) || defined(__AVR_ATmega644P__)
+
+#define digitalPinToPCICR(p)    (((p) >= 0 && (p) <= 31) ? (&PCICR) : ((uint8_t *)0))
+#define digitalPinToPCICRbit(p) (((p) <= 7) ? 1 : (((p) <= 15) ? 3 : (((p) <= 23) ? 2 : (((p) <= 31) ? 3 : 0))))
+#define digitalPinToPCMSK(p)    ( \
+								(	((p) <=  7) ? (&PCMSK1) : \
+								(	((p) <= 15) ? (&PCMSK3) : \
+								(	((p) <= 23) ? (&PCMSK2) : \
+								(	((p) <= 31) ? (&PCMSK0) : \
+									((uint8_t *)0) \
+								)))))
+#define digitalPinToPCMSKbit(p) ( \
+								(	((p) <=  7) ? (p) : \
+								(	((p) <= 15) ? ((p) - 8) : \
+								(	((p) <= 23) ? ((p) - 16) : \
+								(	((p) <= 31) ? ((p) - 24) : \
+								0 )))))
+								
 #elif defined(__AVR_ATmega1280__) || defined(__AVR_ATmega2560__)
 // Specifically for the Arduino Mega 2560 (or 1280 on the original Arduino Mega)
 // A majority of the pins are NOT PCINTs, SO BE WARNED (i.e. you cannot use them as receive pins)
